### PR TITLE
fix(link-with-icon): removed keyboard focus on disable

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)

#3712 

### Description

The main issue that caused the element to still be focused even while disabled. Removed `tabIndex` as it should still respect the focus order while active and not focus during disabled state.

### Changelog

**Removed**

- Removed tabIndex from the `Link` element

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
Hi @IgnacioBecerra I think the internal `<Link>` is keyboard-focusable even without `tabIndex` specified given it uses `<a>` internally. The internal `<Link>` automatically turns off such focusability if it's disabled, by [turning it to `<p>`](https://github.com/carbon-design-system/carbon/blob/v10.20.0/packages/react/src/components/Link/Link.js#L29). That said, would you try simply removing `tabIndex` from the internal `<Link>`? Thanks!

**Comment by IgnacioBecerra** [comment]
@asudoh You're right, Link does turn it into `<p>`, I hadn't noticed that. Removed tabIndex as it's not necessary.

**Comment by jeffchew** [comment]
@IgnacioBecerra looks like you need to update the snapshot file to reflect the change. You can do this with `yarn test:unit:updateSnapshot` in the web components package or react package.

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 1]
```diff
@@ -37,7 +38,7 @@ const LinkWithIcon = ({
       )}
       data-autoid={`${stablePrefix}--link-with-icon`}>
       <Link
-        tabIndex={0}
+        tabIndex={disabled ? -1 : 0}
```
Comment (asudoh) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:41]:
```suggestion
```

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ const { prefix } = settings;
 const LinkWithIcon = ({
   children,
   className: customClassName,
+  disabled,
```
Comment (asudoh) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:24]:
```suggestion
```

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 3]
```diff
@@ -55,6 +56,11 @@ LinkWithIcon.propTypes = {
    */
   children: PropTypes.arrayOf(PropTypes.node),
 
+  /**
+   * Toggles keyboard tab focus for element
+   */
+  disabled: PropTypes.bool,
+
```
Comment (asudoh) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:63]:
```suggestion
```

File: packages/react/src/components/LinkWithIcon/LinkWithIcon.js [hunk 4]
```diff
@@ -78,6 +84,7 @@ LinkWithIcon.propTypes = {
 
 LinkWithIcon.defaultProps = {
   children: [],
+  disabled: false,
```
Comment (asudoh) [inline packages/react/src/components/LinkWithIcon/LinkWithIcon.js:87]:
```suggestion
```

**Review by asudoh — COMMENTED** [review]
Thank you for the update @IgnacioBecerra - Below is what I actually meant in case I wasn't clear enough (my apologies if it's the case). Please let me know if it doesn't work. Thanks!

**Comment by IgnacioBecerra** [comment]
@asudoh Oh, my bad! I forgot to push the changes after writing that comment. No worries, you were clear enough 👍 

Also, I updated the React screenshots, thanks @jeffchew!

**Comment by jeffchew** [comment]
> @asudoh Oh, my bad! I forgot to push the changes after writing that comment. No worries, you were clear enough 👍
> 
> Also, I updated the React screenshots, thanks @jeffchew!

No worries! It looks like you have merge conflicts now, can you resolve?

**Review by asudoh — APPROVED** [review]
Cool, LGTM 👍 once the snapshot is recreated to resolve the merge conflict. Thanks a lot @IgnacioBecerra!

**Comment by jeffchew** [comment]
@IgnacioBecerra you may need to re-run the updateSnapshot task to fix this.

**Pull request merged**